modules/sc-mesh-secure-deployment/src/nats/initd/S9011sNatsMesh

```diff
@@ -10,35 +10,62 @@
 #                    automatically
 ### END INIT INFO
 
+source /opt/mesh-helper.sh
+# sources mesh configuration
+
+RADIO_INDEX=$2   # $2: index of the mesh.conf file to be used e.g. "id0"
+
+source_configuration "${RADIO_INDEX:2}"
+
 DAEMON="mesh-11s_nats.sh"
 DIR="/opt/"
-PIDFILE="/var/run/$DAEMON.pid"
-LOG_FILE=/opt/mesh_11s.log
+_WIFI="${RADIO_INDEX}_MESH_VIF"
+_MODE="${RADIO_INDEX}_MODE"
+PIDFILE="/var/run/${DAEMON}_${RADIO_INDEX}.pid"
+LOG_FILE=/opt/mesh_11s_${RADIO_INDEX}.log
 
-source /opt/mesh-helper.sh
-# sources mesh configuration
-source_configuration
+SYNC_LOCK="/var/run/mesh_start.lock"
 
 # shellcheck source=/dev/null
 [ -r "/etc/default/$DAEMON" ] && . "/etc/default/$DAEMON"
 
 start() {
+    if mkdir $SYNC_LOCK 2>/dev/null; then
+       echo "I have a lock $RADIO_INDEX PID $$"
+    else
+       echo "Instance is already running. Waiting..."
+       # Wait for the lock directory to be removed, indicating the previous instance finished
+       while [ -d "$SYNC_LOCK" ]; do
+         sleep $(( (RANDOM % 5) + 1 ))
+       done
+       mkdir "$SYNC_LOCK"
+       echo "I have a lock $$"
+    fi
+
     printf 'Starting %s: ' "$DAEMON"
-    start-stop-daemon -b -m -S -q -p "$PIDFILE" -a /bin/bash -- -c "exec $DIR$DAEMON mesh > $LOG_FILE 2>&1"
+    if [ "${!_MODE}" == "ap+mesh_scc" ] || [ "${!_MODE}" == "ap+mesh_mcc" ]; then
+      mode=mesh
+    else
+      mode=${!_MODE}
+    fi
+    start-stop-daemon -b -m -S -q -p "$PIDFILE" -a /bin/bash -- -c "exec $DIR$DAEMON $mode $RADIO_INDEX > $LOG_FILE 2>&1"
     status=$?
     if [ "$status" -eq 0 ]; then
         echo "OK"
     else
         echo "FAIL"
     fi
+
+    rmdir "$SYNC_LOCK"
+
     return "$status"
 }
 stop() {
     printf 'Stopping %s: ' "$DAEMON"
-    NEW_PID=$(ps ax | grep -E 'wpa_supplicant\-11s' | awk {'print $1'})
-    kill -9 $NEW_PID
-    rm -fr /var/run/wpa_supplicant/* 2>/dev/null
-    ifconfig "$MESH_VIF" down 2>/dev/null
+    kill -9 "$(cat "$PIDFILE")"
+    kill -9 "$(ps ax | grep -E "[w]pa_supplicant\-11s_$RADIO_INDEX" | awk {'print $1'})"
```

Comment:
Please do not kill -9 (SIGKILL) right away, the correct way is to send SIGTERM first to give programs a chance to finish up and/or exit cleanly while we wait for a bit, then SIGKILL them only if they haven't.

This is another reason to use a proper daemon manager like SystemD to not have to write lots of little paragraphs here and there reinventing the wheel and probably not doing it as well as a modern init system :)

Reply:
I know that we have limitations with init.d and currently we need to live with that.

Reply:
Still need to issue SIGTERM then wait then SIGKILL if necessary 🙂

Otherwise not giving a chance for programs to exit gracefully is likely to create all sorts of unreliability like maybe you won't be able to start another wpa_supplicant or hostapd because the previous one left the interface in a busy state, or at least not without unloading+reloading the driver, maybe rebooting the whole OS or who knows what, much more complicated than just not murdering programs all the time 🙂

Reply:
Not going to fix this now.